data_warehouse/services/training_record_service.py

```diff
@@ -0,0 +1,35 @@
+'''Service for export records'''
+from django.utils.timezone import datetime
+from django.db.models import Q
+from training_record.models import Record
+
+
+class TrainingRecordService:
+    '''Service for export records'''
+    @staticmethod
+    def get_records(user, event_name=None,
+                    event_location=None, start_time=None, end_time=None):
+        '''get matched records'''
+        if event_name is None:
```

Comment:
I think the validation logic can be organized into a serializer which requires no extra efforts but improves the readability